src/prefect/tasks.py

```diff
@@ -1557,7 +1554,7 @@ def delay(self, *args: P.args, **kwargs: P.kwargs) -> PrefectDistributedFuture:
         return self.apply_async(args=args, kwargs=kwargs)
 
     @sync_compatible
-    async def serve(self) -> NoReturn:
+    async def serve(self) -> None:
```

Comment:
Is `NoReturn` more accurate here?

Reply:
pyright was complaining about it, but reverted for now until i deal with `sync_compatible` here